tests/scattering1d/test_correctness.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+import numpy as np
+import scipy.signal
+from kymatio.numpy import Scattering1D
+
+
+# set True to execute all test functions without pytest
+run_without_pytest = 0
+
+
+def test_T():
+    """Test that `T` controls degree of invariance as intended."""
+    # configure scattering & signal
+    J = 11
+    Q = 16
+    N = 4096
+    width = N//8
+    shift = N//8
+    T0, T1 = N//2, N//4
+    freq_fracs = (4, 8, 16, 32)
+
+    # make signal & shifted
+    window = scipy.signal.tukey(width, alpha=0.5)
+    window = np.pad(window, (N - width) // 2)
+    t  = np.linspace(0, 1, N, endpoint=False)
+    x  = np.sum([np.cos(2*np.pi * N/ff * t) for ff in freq_fracs], axis=0
+                ) * window
+    xs = np.roll(x, shift)
+
+    # make scattering objects
+    ts0 = Scattering1D(J=J, Q=Q, shape=N, T=T0, average=1, out_type='array')
+    ts1 = Scattering1D(J=J, Q=Q, shape=N, T=T1, average=1, out_type='array')
+
+    # scatter
+    ts0_x  = ts0.scattering(x)
+    ts0_xs = ts0.scattering(xs)
+    ts1_x  = ts1.scattering(x)
+    ts1_xs = ts1.scattering(xs)
+
+    # compare distances
+    l1l2_00_xxs = l1l2(ts0_x, ts0_xs)
+    l1l2_11_xxs = l1l2(ts1_x, ts1_xs)
+
+    th0 = .13
+    th1 = th0 * 2
+    assert l1l2_00_xxs < th0, "{} > {}".format(l1l2_00_xxs, th0)
+    assert l1l2_11_xxs > th1, "{} < {}".format(l1l2_11_xxs, th1)
+
+
+def _l1l2(x):
+    return np.sum(np.sqrt(np.mean(x**2, axis=1)), axis=0)
+
+def l1l2(x0, x1):
+    """Coeff distance measure; Thm 2.12 in https://arxiv.org/abs/1101.2286"""
+    return _l1l2(x1 - x0) / _l1l2(x0)
+
+
+if __name__ == '__main__':
```

Comment:
Remove.

Reply:
Why? This is for local debugging.

Reply:
Then run `pytest tests/scattering1d/test_correctness.py`.

Reply:
That doesn't permit interactive debugging.

Reply:
Then run `pytest --trace`, `pytest --pdb`, or add those lines in when you're debugging. At any rate, this doesn't belong in the repo.

Reply:
Alright. Will remove last before merge, I might still use it till then.

Reply:
this was resolved

Reply:
@pawarrick Not exactly

Reply:
@OverLordGoldDragon do you mean the point about interactive debugging?  I can run pytest in my IDE fine without main scaffolding.  In any event, this should not get in the way of the merge in my opinion.